xbmc/peripherals/bus/PeripheralBus.h

```diff
@@ -54,7 +54,7 @@ namespace PERIPHERALS
     /*!
      * @return True if this bus needs to be polled for changes, false if this bus performs updates via callbacks
      */
-    bool NeedsPolling(void) const { return m_bNeedsPolling; }
+    bool NeedsPolling(void) const { CSingleLock lock(m_critSection); return m_bNeedsPolling; }
```

Comment:
Std::atomic are implemented using cas and does not require locking for reading or writing to. Locking is still required to synchronize a group of operations.  Using std::atomic in this code will create temp bool, cas into it and return the temp bool